rustworkx-core/src/connectivity/minimum_cycle_basis.rs

```diff
@@ -0,0 +1,493 @@
+            // Add cross edges with weight
+            gi.add_edge(
+                gi_u,
+                NodeIndex::new(gi_name_to_node_index[&v_lifted_name]),
+                weight,
+            );
+            gi.add_edge(
+                NodeIndex::new(gi_name_to_node_index[&u_lifted_name]),
+                gi_v,
+                weight,
+            );
+        } else {
+            // Add in-plane edges with weight
+            gi.add_edge(
+                NodeIndex::new(gi_name_to_node_index[&u_name]),
+                NodeIndex::new(gi_name_to_node_index[&v_name]),
+                weight,
+            );
+            gi.add_edge(
+                NodeIndex::new(gi_name_to_node_index[&u_lifted_name]),
+                NodeIndex::new(gi_name_to_node_index[&v_lifted_name]),
+                weight,
+            );
+        }
+    }
+    // Instead of finding the shortest path between each node and its lifted node, store the shortest paths in a list to find the shortest paths among them
+    let mut shortest_path_map: HashMap<String, i32> = HashMap::new();
+    for nodeid in graph.node_identifiers() {
+        let node_weight = graph.node_weight(nodeid).unwrap();
+        let node_name = format!("{:?}", node_weight).trim_matches('"').to_string();
+        let lifted_node_name = format!("{}_lifted", node_name);
+        let node = gi_name_to_node_index[&node_name];
+        let nodeidx = NodeIndex::new(node);
+        let lifted_node = gi_name_to_node_index[&lifted_node_name];
+        let lifted_nodeidx = NodeIndex::new(lifted_node);
+        let result = astar(
+            &gi,
+            nodeidx,
+            |finish| finish == lifted_nodeidx,
+            |e| *e.weight(),
+            |_| 0,
+        );
+        match result {
+            Some((cost, _path)) => {
+                shortest_path_map.insert(node_name, cost);
+            }
+            None => {}
+        }
+    }
+    let min_start = shortest_path_map
+        .keys()
+        .min_by_key(|k| &shortest_path_map[k.as_str()])
+        .unwrap();
+    let min_start_node_index = gi_name_to_node_index[min_start];
+    let min_start_lifted_node_index = gi_name_to_node_index[&format!("{}_lifted", min_start)];
+    let min_start_node = NodeIndex::new(min_start_node_index);
+    let min_start_lifted_node = NodeIndex::new(min_start_lifted_node_index);
+    let result = astar(
+        &gi,
+        min_start_node,
+        |finish| finish == min_start_lifted_node,
+        |e| *e.weight(),
+        |_| 0,
+    );
+    // Store the shortest path in a list and translate lifted nodes to original nodes
+    let mut min_path: Vec<usize> = Vec::new();
+    match result {
+        Some((_cost, path)) => {
+            for node in path {
+                let node_name = gi.node_weight(node).unwrap();
+                if node_name.contains("_lifted") {
+                    let original_node_name = node_name.replace("_lifted", "");
```

Comment:
If you need to add additional information to the hash you can also hash tuples in Rust. Or you can come up with a custom struct and write a hasher for it